lib/auth/v4/queryAuthCheck.js

```diff
@@ -23,6 +24,13 @@ function check(request, log, data) {
     if (Object.keys(authParams).length !== 5) {
         return { err: errors.InvalidArgument };
     }
+
+    const token = data['X-Amz-Security-Token'];
```

Comment:
It's not obvious to me that AWS (hence our implementation as well) expects query params to be case sensitive, while headers are not wrt HTTP spec, so I don't think a comment would hurt (not necessarily on this particular header though).

This being said, it's fine for me if you don't feel the need for a comment.